### ISSUE TITLE
Loading external files?

### ISSUE THREAD

**Issue by Neilt100** [modflowpy/flopy#175]
I've been using flopy to load a disu file. As an example I tried to load the biscayne disu file, included in the USG quadtree example.

When I try to load the disu file I  get an error:

```
    disu = flopy.modflow.ModflowDisU.load('biscayne.disu', mf)
 File "C:\Python27\lib\site-packages\flopy\modflow\mfdisu.py", line 494, in load
    ext_unit_dict)
  File "C:\Python27\lib\site-packages\flopy\utils\util_array.py", line 2513, in load
    assert cr_dict['nunit'] in list(ext_unit_dict.keys())
AttributeError: 'NoneType' object has no attribute 'keys'
```

I think the problem is related to the header line for the 'top' array in the biscayne disu file:
       11          1           (5G13.0)         -1 TOP(K=1)

Flopy seems to assume that the file type is 'external'.
If I change the header line to:
internal          1           (5G13.0)         -1 TOP(K=1)

The top array is read in correctly.
Is this an issue when the unit number is the same as the file it is reading?

Thanks.

**Comment by langevin-usgs** [comment]
@Neilt100 can you tell me what version of flopy you are using?  Are you using the develop version?  I'll try to take a look at this soon.  Thanks for reporting the issue.

**Comment by Neilt100** [comment]
Hi, I'm using the development version of flopy- v3.2.6 running under windows. 

Cheers.

**Comment by langevin-usgs** [comment]
@Neilt100, so it turns out that it is not really a bug, but rather an issue with the fact that the DISU package for the Biscayne model is using the older style array control records (combined with our limited support for MODFLOW-USG).  When a file has those older style control records, the load methods need more information in order to match unit numbers with files.  This is done by constructing an external unit dictionary (called ext_unit_dict).  The following hack shows how to do this in flopy.

```
import flopy

fname = 'biscayne.disu'
fhandle = open(fname, 'r')
packages = []
ext_unit_dict = {11: flopy.utils.mfreadnam.NamData('DISU', fname, fhandle, packages)}

mf = flopy.modflow.Modflow(version='mfusg', verbose=True)
disu = flopy.modflow.ModflowDisU.load(fhandle, mf, ext_unit_dict=ext_unit_dict)
```

I'll start thinking about the next step toward more robust USG support.

**Comment by Neilt100** [comment]
Many thanks for the workaround. That closes this issue out for me. Cheers.

**Comment by langevin-usgs** [comment]
@Neilt100, I made some progress with the loading of unstructured models.  You can now use the following simple script to load the unstructured biscayne model.  It is still pretty fragile at this point, but I think I've settled on how flopy will need to handle unstructured models in general.  Note that you need to set check=False to work as checking needs to be revised.

```
import flopy
mf = flopy.modflow.Modflow.load('biscayne.nam', version='mfusg', verbose=True,
                                check=False)
```

When you run the script, you should get something like this:

```
   The following 7 packages were successfully loaded.
      biscayne.disu
      biscayne.bas
      biscayne.oc
      biscayne.lpf
      biscayne.riv
      biscayne.wel
      biscayne.sms
   The following 3 packages were not loaded.
      biscayne.output.list
      biscayne.rch
      biscayne.evt
```

So basically, the model load will now be able to load DISU, BAS, OC, LPF, SMS, and the list based boundary packages.  RCH and EVT don't work yet, nor do some of the options in LPF.  Plotting, reading binary files, etc. also need to be implemented.

See commit https://github.com/modflowpy/flopy/commit/578af96e2e580cfbdcbb50499e4c2b33e37c971d additional details.
